content/docs/guides/uninstall.md

```diff
@@ -124,24 +125,43 @@ namespace "<namespace>" deleted
+Uninstalling OSM (through the `osm uninstall mesh` command or through Helm) will uninstall osm control plane components.
+However, it will leave behind certain OSM resources to prevent unintended consequences for the cluster after uninstalling OSM.
+The resources that are left behind will depend on whether OSM was uninstalled from a managed or unmanaged cluster environment.
 
-If there are no other service meshes running in the same cluster and the required custom resources have been backed up, the CRDs can be removed from the cluster using `kubectl`.
+`osm uninstall mesh` uninstalls the following resources in both unmanaged and managed environments:
+1. remove osm control plane components, including control plane pods,
+2. remove/un-patch the conversion webhook fields from all the CRDs (which OSM adds to support multiple CR versions),
 
-Run the following `kubectl` commands:
+In addition, running `osm uninstall mesh` will uninstall the following additional resources, but only if OSM is being uninstalled from an _unmanaged_ environment. These resources will be left behind when osm is uninstalled from a _managed_ environment.
+1. the osm resources such as meshconfig, RBAC, control plane deployments, and control plane services,
+2. osm control plane secrets,
+3. osm mutating webhook configurations,
+4. osm validating webhook configurations,
 
-```console
-kubectl delete crd meshconfigs.config.openservicemesh.io
-kubectl delete crd multiclusterservices.config.openservicemesh.io
-kubectl delete crd egresses.policy.openservicemesh.io
-kubectl delete crd ingressbackends.policy.openservicemesh.io
-kubectl delete crd httproutegroups.specs.smi-spec.io
-kubectl delete crd tcproutes.specs.smi-spec.io
-kubectl delete crd traffictargets.access.smi-spec.io
-kubectl delete crd trafficsplits.split.smi-spec.io
-```
+OSM ensures that all the CRDs mentioned [here](https://github.com/openservicemesh/osm/tree/{{< param osm_branch >}}/cmd/osm-bootstrap/crds)
+exist in the cluster at install time.
+During installation, if they are not already installed, the `osm-bootstrap` pod will install them before the rest of the control plane components are running.
+This is the same behavior when using the Helm charts to install OSM as well.
+When uninstalling osm, both the `osm uninstall mesh` command and Helm uninstallation will not delete any OSM or SMI CRD in any cluster environment (managed and unmanaged) for primarily two reasons:
+1. CRDs are cluster-wide resources and may be used by other service meshes or resources running in the same cluster,
+2. deletion of a CRD will cause all custom resources corresponding to that CRD to also be deleted.
+
+> Note: `osm uninstall mesh` does not delete the namespace where osm is installed. This prevents unintended consequences, such as if osm was installed in `kube-system`, as deleting that namespace may delete important cluster resources and may have unintended consequences.
```

Comment:
remove the note, this is added too many time and everywhere. We just need it once and in one place